acdc-ws/conf/application.conf

```diff
@@ -15,6 +15,16 @@
 # HOCON will fall back to substituting environment variable:
 #mykey = ${JAVA_HOME}
 
+acdc.auth = {
+    enabled = true
+    header-name = "x-api-key"
+    hashed-keys = {
+        "f6e42a3c0dffee079face0da061ee2c9a871eebe098ac481248e34cfe023955b" = ["admin", "user"]
```

Comment:
let's get those from ENV

Reply:
ok.  new commit pushed to get from MCE_ENV_X_API_KEYS

I wrote a regex parser function in AuthorizationSettings to handle the string from ENV, string cannot be cast to com.typesafe.config.ConfigList

Reply:
how about `hashed-keys = { ${X_API_KEYS} = ["admin", "user]] }`? then you don't need to parse them and user can overwrite those pretty easily

Reply:
HOCON substitution seems to work on the value side only for the key=value format.  We can swap the role and identity key around like below.  
```
    hashed-keys = {
        admin = [ ${MCE_ENV_X_API_ADMIN1}, ${MCE_ENV_X_API_ADMIN2} ]
        user = [ ${MCE_ENV_X_API_USER1}, ${MCE_ENV_X_API_USER2} ]
    }	
```
We probably do not have to deal with a long list of identities, just a few for identity keys rotation.  

The swapped format looks different wrt IAM conventions where an identity key is allowed one or more role access, with the allowed roles on the value side.  

The regex parse is done, I have tested it with newline character, whitespace etc.  Dev-ops would have to manage an entire map of identity hashed key / role access as a potentially long string ENV variable.  There is also the alternative if dev-ops edit the conf file itself.  

What do you think?

Reply:
I think those envs are mainly to give an example of those configs, and potentially make it easy for testing. In production, one would most likely always going to supply their own configs to overwrite some configurations and chooses what's best for the situation. If we cannot have envs in the key, it makes sense to swap the role and identity.

Reply:
new commit pushed.  It uses the role as the key as discussed above.